src/strategies/npm-audit.js

```diff
@@ -14,18 +23,60 @@ export function NPMAuditStrategy() {
   };
 }
 
+async function checkIfProjectUsePnpm(path) {
+  return new Promise((resolve) => {
+    fs.access(`${path}/pnpm-lock.yaml`, fs.constants.F_OK, (err) => {
+      if (err) {
+        resolve(false);
+      } else {
+        console.log("Project use pnpm");
+        resolve(true);
+      }
+    });
+  });
+}
+
+async function launchPnpmAudit(path) {
+  console.log("launch Pnpm Audit");
+  const opts = { 
+    include: { dependencies: true, devDependencies: true, optionalDependencies: false },
+    lockfileDir: path,
+    registry: "https://registry.npmjs.org",
+  };
+  
+  readWantedLockfile(path, {})
+    .then((lockfile) => { 
+      console.log("Lockfile ok");
+      audit(lockfile, { registry: "https://registry.npmjs.org" }, opts); })
+    .then((auditResult) => {
+      console.log("Audit result -> ", JSON.stringify(auditResult, null, 2));
+
+      return auditResult; 
+    })
+    .catch((err) => {
+      console.log("Error -> ", err);
+      throw err;
+    });
```

Comment:
Why using legacy Promise syntax in Async function? Please use async/await

Reply:
Thanks for the review! Indeed it is an oversight. I copied and pasted their example without modifying it. I will refactor and push cleaner code. :)